plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -293,6 +239,51 @@ protected void checkInformationSchemaViewsForMaterializedView(String schemaName,
                 .hasMessageFindingMatch("(?s)Expecting.*to contain:.*\\Q[(" + viewName + ")]");
     }
 
+    // Override is required because metadata deletes do not return the number of rows deleted. See todo in IcebergMetadata#executeDelete
+    @Test
+    @Override
+    public void testDeleteWithComplexPredicate()
+    {
+        skipTestUnlessSupportsDeletes();
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete_complex_", "AS SELECT * FROM orders")) {
+            // delete half the table, then delete the rest
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE orderkey % 2 = 0", "SELECT count(*) FROM orders WHERE orderkey % 2 = 0");
+            assertQuery("SELECT * FROM " + table.getName(), "SELECT * FROM orders WHERE orderkey % 2 <> 0");
+
+            query("DELETE FROM " + table.getName());
+            assertQuery("SELECT * FROM " + table.getName(), "SELECT * FROM orders LIMIT 0");
+
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE rand() < 0", 0);
+        }
+    }
+
+    // Override is required because metadata deletes do not return the number of rows deleted. See todo in IcebergMetadata#executeDelete
+    @Test
+    @Override
+    public void testDeleteWithSubquery()
+    {
+        skipTestUnlessSupportsDeletes();
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete_subquery", "AS SELECT * FROM nation")) {
+            // delete using a subquery
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE regionkey IN (SELECT regionkey FROM region WHERE name LIKE 'A%')", 15);
+            assertQuery(
+                    "SELECT * FROM " + table.getName(),
+                    "SELECT * FROM nation WHERE regionkey IN (SELECT regionkey FROM region WHERE name NOT LIKE 'A%')");
+        }
+
+        // TODO (https://github.com/trinodb/trino/issues/5901) Use longer table name once Oracle version is updated
```

Comment:
Oracle-related comment is not applicable to an Iceberg test class. Simply remove the TODO comment from here.